tests/e2e/test_create_preview_collection.py

```diff
@@ -558,12 +592,14 @@ def task_check_your_answers(
 ):
     check_your_answers_page = RunnerCheckYourAnswersPage(tasklist_page.page, tasklist_page.domain, grant_name)
 
-    for question_to_test in questions_to_test.values():
-        if question_to_test["type"] == "group":
-            for nested_question in question_to_test["questions"]:
-                assert_check_your_answers(check_your_answers_page, nested_question)
-        else:
-            assert_check_your_answers(check_your_answers_page, question_to_test)
+    def _assert_check_your_answers(questions_to_check: list[TQuestionToTest]):
+        for question_to_check in questions_to_check:
+            if question_to_check["type"] == "group":
+                _assert_check_your_answers(question_to_check["questions"])
+            else:
+                assert_check_your_answers(check_your_answers_page, question_to_check)
+
+    _assert_check_your_answers(list(questions_to_test.values()))
```

Comment:
Can we move this outside of the function somewhere? And give it a better name? Having a function called `assert_view_report_answers` and `_assert_view_report_answers` feels a bit confusing. I think we probably want to take a step back and refactor to make sure things are feeling simple and clean and clear.

Reply:
Yeah, fair point. I think add another will further complicate so will look to refactor then?